src/widgets.ts

```diff
@@ -93,6 +128,20 @@ export abstract class Widget<
     this.teardownIframe()
   }
 
+  /**
+   * Uses matching to get our url targetOrigin, or falls back to our widgets url
+   */
+  private getTargetOrigin() {
+    let targetOrigin
+    const baseUrlPattern = /^https?:\/\/[^/]+/i
+
+    if (this.options.url && this.options.url.match(baseUrlPattern)) {
```

Comment:
Keep in mind that `this.options.url` won't always be set. It won't be there if a client is using the proxy settings (or if using the sso token which doesn't exists now but will in the future)

I think we're gonna want to save the URL we're loading somewhere in https://github.com/mxenabled/web-widget-sdk/blob/a2e3930a34968184a71646b5a2b88e8bf0704d7b/src/widgets.ts#L149-L166 instead.

Reply:
If they are using a proxy, should we be getting the `targetOrigin` from the proxy url? Or just continue to fall back to our "https://widgets.moneydesktop.com" url?

Reply:
@minond -- Let's chat about this, I have a few questions.

Reply:
This has been updated, please take a look when you can.

Reply:
Looking good.